libdnf/module/module_db.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_DB_HPP
+#define LIBDNF_MODULE_MODULE_DB_HPP
+
+#include "system/state.hpp"
+
+#include "libdnf/base/base_weak.hpp"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace libdnf::module {
+
+
+class ModuleDB {
+public:
+    ModuleDB(BaseWeakPtr base) : base(base) {}
```

Comment:
We could take a const reference `ModuleDB(const BaseWeakPtr & base)` instead of a copy.